### ISSUE TITLE
Synchronize more than one folder

### ISSUE THREAD

**Issue by pchico83** [okteto/okteto#668]
**Is your feature request related to a problem? Please describe.**

`okteto up` synchronizes the local folder where the `okteto.yaml` is placed on to the destination path defined by the  `mountpath` or `workdir` field. 

There are several uses cases where you want to synchronize a different folder than the one the `okteto.yaml` is placed on, or where you want to synchronize several folders. The current `okteto up` behavior is not flexible enough.

**Describe the solution you'd like**
I'd like to be able to specify multiple synchronization points. 

I propose the field `sync` to specify the synching behavior. `mounthpath` and `subpath` will be deprecated, and `workdir` keeps the current behavior if `sync`  is not defined. In the future, it will also be deprecated.

`sync` is a list of `localPath:remotePath` pairs. Each pair would indicate a local folder to synchronize into a remote destination.

`LocalPath` can be:
- relative path: In which case, the base folder is the location of `okteto.yml`
- absolute path

We can also support `env var` expansion in the paths, in case a user wants to synchronize ssh keys, env files, etc...

**example**

```
name: hello-world
image: okteto/golang:1
sync: 
- code: /code
- assets: /usr/html/public/assets
- $HOME/.npm/secret:/root/.npm/secret
command: bash
```

This gives the ability to organize files as the user prefers.

**Comment by rberrelleza** [comment]
Currently, okteto will start the synchronizing from the folder where the `okteto.yaml` is placed on. I can think of a couple of ways to get to this:
1. Do the sync from where `okteto` was invoked, independently of where the manifest physically resides. 
1. Add a new parameter to the manifest to specify tell synchronization folder. This is non-breaking, but it would have to allow for relative paths, since we want the `okteto.yaml` to not have any dependencies on local paths.

**Comment by rberrelleza** [comment]
For relative paths, should we use `pwd` as the base, or the manifest's location?

**Comment by pchico83** [comment]
> For relative paths, should we use `pwd` as the base, or the manifest's location?

the manifest's location. if not, the behavior of the okteto manifest is also relative, which is not a consistent behavior

**Comment by sameera466** [comment]
Hi, 
I have a similar use case too. All of my services are part of different repos (I work on integration) and all kubernetes yamls are collected as part of another repo. It would really help if i could add okteto manifests to another repo and point to different folders to synchronise

**Comment by tungphuong** [comment]
Did you have any updates? When do you release this feature? I also have the same issue when need to sync share source code folders

**Comment by pchico83** [comment]
@tungphuong we don't have an estimate yet. As a workaround, you can place your different okteto manifests on the root folder, and run `okteto up -f manifest1.yml`. Would that work?

**Comment by pchico83** [comment]
FYI, we have added this issue to the august roadmap

**Comment by pchico83** [comment]
This feature is available from Okteto CLI version 1.8.17.
Full docs are available here:
https://okteto.com/docs/reference/manifest#sync-string-required
